### PR TITLE
Adds integration test for changing signing algorithm

### PR DESCRIPTION
**Pull request by mitchhentges**

### CODE REVIEW

File: signingscript/test/integration/test_autograph.py [hunk 1]
```diff
@@ -217,25 +207,50 @@ def _extract_compress_type_per_filename(path):
...
+
+    algorithm_line = [line for line in command.stdout.split('\n') if 'Signature algorithm:' in line][0]
+    algorithm = algorithm_line.strip().split(' ')[2]
+    return algorithm[:-1]  # remove comma at end of algorithm
+
+
+@pytest.mark.asyncio
+@skip_when_no_autograph_server
+@pytest.mark.parametrize('format,expected_algorithm', ((
+    'autograph_apk_fennec_sha1', 'SHA1withSHA1withRSA',
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: Add TODO instructions to change Fennec algo to `RSA` alone. This misconfig caused https://bugzilla.mozilla.org/show_bug.cgi?id=1513564. There are several OIDs and we need that one https://github.com/mozilla-services/pkcs7/blob/3cffc6fbfe837a6fe43e8e336153379a69ef02bd/pkcs7.go#L66 on Fennec

File: signingscript/test/integration/test_autograph.py [hunk 2]
```diff
@@ -217,25 +207,50 @@ def _extract_compress_type_per_filename(path):
...
+    log.info("running {}".format(cmd))
+    command = subprocess.run(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True
+    )
+
+    assert command.returncode == 0
+
+    algorithm_line = [line for line in command.stdout.split('\n') if 'Signature algorithm:' in line][0]
+    algorithm = algorithm_line.strip().split(' ')[2]
+    return algorithm[:-1]  # remove comma at end of algorithm
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: Nit: 
```suggestion
   
     return algorithm.rstrip(',')  # remove comma at end of algorithm
```

File: signingscript/test/integration/test_autograph.py [hunk 3]
```diff
@@ -173,7 +171,7 @@ def _get_java_path(tool_name):
     return tool_name
 
 
-def _instanciate_keystore(keystore_path, certificate_path, certificate_alias):
+def _instantiate_keystore(keystore_path, certificate_path, certificate_alias):
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: :+1:

File: signingscript/test/integration/test_autograph.py [hunk 4]
```diff
@@ -217,25 +207,50 @@ def _extract_compress_type_per_filename(path):
...
-    for signature_file in ('META-INF/SIGNATURE.RSA', 'META-INF/SIGNATURE.SF', 'META-INF/MANIFEST.MF'):
-        del zip_infos_after_signature[signature_file]
+    context.signing_servers = {'project:releng:signing:cert:dep-signing': [SigningServer(
+        *["http://localhost:5500", "bob", "1234567890abcdefghijklmnopqrstuvwxyz1234567890abcd", [format], "autograph"]
+    )]}
+    context.task = _craft_task([file_name], signing_format=format)
 
-    # We want to make sure compression type hasn't changed after the signature
-    # https://github.com/mozilla-services/autograph/issues/164
-    assert zip_infos_before_signature == zip_infos_after_signature
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: We probably want to keep this test. It caused https://bugzilla.mozilla.org/show_bug.cgi?id=1513870#c22 last week.

File: signingscript/test/integration/test_autograph.py [hunk 5]
```diff
@@ -1,17 +1,15 @@
-import aiohttp
-import asyncio
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: :+1:

File: signingscript/test/integration/test_autograph.py [hunk 6]
```diff
@@ -217,25 +207,50 @@ def _extract_compress_type_per_filename(path):
...
+    algorithm = algorithm_line.strip().split(' ')[2]
+    return algorithm[:-1]  # remove comma at end of algorithm
+
+
+@pytest.mark.asyncio
+@skip_when_no_autograph_server
+@pytest.mark.parametrize('format,expected_algorithm', ((
+    'autograph_apk_fennec_sha1', 'SHA1withSHA1withRSA',
+), (
+    'autograph_apk_fennec', 'SHA256withSHA256withRSA',
```
Comment (JohanLorenzo) [inline signingscript/test/integration/test_autograph.py]: Add TODO instruction to say we should verify the digest algorithm once SHA1 is not automatically added anymore

File: signingscript/test/integration/test_autograph.py (hunk 4 above)
Comment (mitchhentges) [reply signingscript/test/integration/test_autograph.py]: :grimacing: that's a huge :yikes: from me, I'm not sure how this got removed.
Great eye, and I promise it wasn't deliberate (is that worse?)

Comment (JohanLorenzo) [reply signingscript/test/integration/test_autograph.py]: Don't blame yourself. This is why code reviews exist :)

File: signingscript/test/integration/test_autograph.py (hunk 6 above)
Comment (mitchhentges) [reply signingscript/test/integration/test_autograph.py]: Is this the same todo as from [here](https://github.com/mozilla-releng/signingscript/pull/96#discussion_r243585396)?

Comment (JohanLorenzo) [reply signingscript/test/integration/test_autograph.py]: Nope. These are 2 different TODOs. The other comment is for the signature algorithm. Please let me know if you need more details

**Pull request merged**